internal/resolution/entitysources/catalogdsource.go

```diff
@@ -5,67 +5,66 @@ import (
 	"encoding/json"
 	"fmt"
 
+	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
 	"github.com/operator-framework/deppy/pkg/deppy"
 	"github.com/operator-framework/deppy/pkg/deppy/input"
-	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/entity"
 	"github.com/operator-framework/operator-registry/alpha/property"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/entity"
 )
 
-// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// CatalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
 // made accessible on-cluster by https://github.com/operator-framework/catalogd.
 // It is an implementation of deppy defined input.EntitySource
-type catalogdEntitySource struct {
+type CatalogdEntitySource struct {
 	client client.Client
 }
 
-func NewCatalogdEntitySource(client client.Client) *catalogdEntitySource {
-
-	return &catalogdEntitySource{client: client}
+func NewCatalogdEntitySource(client client.Client) *CatalogdEntitySource {
+	return &CatalogdEntitySource{client: client}
 }
 
-func (es *catalogdEntitySource) Get(ctx context.Context, id deppy.Identifier) (*input.Entity, error) {
+func (es *CatalogdEntitySource) Get(_ context.Context, _ deppy.Identifier) (*input.Entity, error) {
 	panic("not implemented")
 }
 
-func (es *catalogdEntitySource) Filter(ctx context.Context, filter input.Predicate) (input.EntityList, error) {
+func (es *CatalogdEntitySource) Filter(ctx context.Context, filter input.Predicate) (input.EntityList, error) {
 	resultSet := input.EntityList{}
 	entities, err := getEntities(ctx, es.client)
 	if err != nil {
 		return nil, err
 	}
-	for _, entity := range entities {
-		if filter(&entity) {
-			resultSet = append(resultSet, entity)
+	for i := range entities {
+		if filter(&entities[i]) {
```

Comment:
gosec: G601: Implicit memory aliasing in for loop.

Reply:
+1 on this fix. However it makes me wonder... is it necessary or important that `input.Predicate` takes a pointer to an `input.Entity`? Could we take a `input.Entity` rather than an `*input.Entity`?